src/transformStateWithClones.js

```diff
@@ -8,6 +8,40 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const statefulClone = {
+    ...state,
+  };
+  const statefulCloneCloned = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties':
+        for (const key in actions[i].extraData) {
+          statefulClone[key] = actions[i].extraData[key];
+        }
+        break;
+
+      case 'removeProperties':
+        for (let y = 0; y < actions[i].keysToRemove.length; y++) {
+          delete statefulClone[actions[i].keysToRemove[y]];
+        }
+        break;
+
+      case 'clear':
```

Comment:
Consider using strict equality `===` instead of loose equality `==` when comparing object length. 🤔